### DISCUSSION TITLE
Using a shader to draw a 2D grid to screen?

### DISCUSSION THREAD

**Discussion by Liam19** [bevyengine/bevy#9938] [Q&A] [answered]
Im trying to draw a simple infinite 2D grid to the screen. The grid should translate and scale based on the position and zoom of the Camera2D.

What kind of shader should I use? 
Currently im using a mesh entity that follows the view, then use a fragment shader to draw the grid on the mesh. This works but the grid is static because i dont know how to send the camera's position and zoom to the shader.
Do i need to use a compute shader?

**Comment by rlidwka** [comment, marked as answer]
i'll let someone else to answer actual question, but you should probably check out this plugin that does something very similar (maybe check out its source code):
https://github.com/ForesightMiningSoftwareCorporation/bevy_infinite_grid
